src/java.base/share/classes/java/lang/Long.java

```diff
@@ -446,36 +448,83 @@ private static void formatUnsignedLong0UTF16(long val, int shift, byte[] buf, in
-            formatUnsignedLong0(msb >>> 16, 4, buf, 9,  4);
-            formatUnsignedLong0(msb >>> 32, 4, buf, 0,  8);
 
-            buf[23] = '-';
-            buf[18] = '-';
+            ByteArray.setLong(buf, 0, ((long) i0 << 48) | ((long) i1 << 32) | ((long) i2 << 16) | i3);
+            buf[8] = '-';
+            ByteArray.setInt(buf, 9, (i4 << 16) | i5);
             buf[13] = '-';
-            buf[8]  = '-';
+            ByteArray.setInt(buf, 14, (i6 << 16) | i7);
+            buf[18] = '-';
+            ByteArray.setInt(buf, 19, (i8 << 16) | i9);
+            buf[23] = '-';
+            ByteArray.setLong(buf, 24, ((long) i10 << 48) | ((long) i11 << 32) | ((long) i12 << 16) | i13);
+            ByteArray.setInt(buf, 32, (i14 << 16) | i15);
 
             return new String(buf, LATIN1);
-        } else {
-            byte[] buf = new byte[72];
-
-            formatUnsignedLong0UTF16(lsb,        4, buf, 24, 12);
-            formatUnsignedLong0UTF16(lsb >>> 48, 4, buf, 19, 4);
-            formatUnsignedLong0UTF16(msb,        4, buf, 14, 4);
-            formatUnsignedLong0UTF16(msb >>> 16, 4, buf, 9,  4);
-            formatUnsignedLong0UTF16(msb >>> 32, 4, buf, 0,  8);
-
-            StringUTF16.putChar(buf, 23, '-');
-            StringUTF16.putChar(buf, 18, '-');
-            StringUTF16.putChar(buf, 13, '-');
-            StringUTF16.putChar(buf,  8, '-');
-
-            return new String(buf, UTF16);
         }
+
+        byte[] buf = new byte[72];
+        int off = ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN ? 1 : 0;
+
+        buf[0 + off] = (byte) (i0 >> 8);
+        buf[2 + off] = (byte) i0;
+        buf[4 + off] = (byte) (i1 >> 8);
+        buf[6 + off] = (byte) i1;
+        buf[8 + off] = (byte) (i2 >> 8);
+        buf[10 + off] = (byte) i2;
+        buf[12 + off] = (byte) (i3 >> 8);
+        buf[14 + off] = (byte) i3;
+        buf[16 + off] = '-';
+        buf[18 + off] = (byte) (i4 >> 8);
+        buf[20 + off] = (byte) i4;
+        buf[22 + off] = (byte) (i5 >> 8);
+        buf[24 + off] = (byte) i5;
+        buf[26 + off] = '-';
+        buf[28 + off] = (byte) (i6 >> 8);
+        buf[30 + off] = (byte) i6;
+        buf[32 + off] = (byte) (i7 >> 8);
+        buf[34 + off] = (byte) i7;
+        buf[36 + off] = '-';
+        buf[38 + off] = (byte) (i8 >> 8);
+        buf[40 + off] = (byte) i8;
+        buf[42 + off] = (byte) (i9 >> 8);
+        buf[44 + off] = (byte) i9;
+        buf[46 + off] = '-';
+        buf[48 + off] = (byte) (i10 >> 8);
+        buf[50 + off] = (byte) i10;
+        buf[52 + off] = (byte) (i11 >> 8);
+        buf[54 + off] = (byte) i11;
+        buf[56 + off] = (byte) (i12 >> 8);
+        buf[58 + off] = (byte) i12;
+        buf[60 + off] = (byte) (i13 >> 8);
+        buf[62 + off] = (byte) i13;
+        buf[64 + off] = (byte) (i14 >> 8);
+        buf[66 + off] = (byte) i14;
+        buf[68 + off] = (byte) (i15 >> 8);
+        buf[70 + off] = (byte) i15;
+
+        return new String(buf, UTF16);
```

Comment:
Maybe restore the `else` block?
```suggestion
        } else {
            byte[] buf = new byte[72];
            int off = ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN ? 1 : 0;

            buf[0 + off] = (byte) (i0 >> 8);
            buf[2 + off] = (byte) i0;
            buf[4 + off] = (byte) (i1 >> 8);
            buf[6 + off] = (byte) i1;
            buf[8 + off] = (byte) (i2 >> 8);
            buf[10 + off] = (byte) i2;
            buf[12 + off] = (byte) (i3 >> 8);
            buf[14 + off] = (byte) i3;
            buf[16 + off] = '-';
            buf[18 + off] = (byte) (i4 >> 8);
            buf[20 + off] = (byte) i4;
            buf[22 + off] = (byte) (i5 >> 8);
            buf[24 + off] = (byte) i5;
            buf[26 + off] = '-';
            buf[28 + off] = (byte) (i6 >> 8);
            buf[30 + off] = (byte) i6;
            buf[32 + off] = (byte) (i7 >> 8);
            buf[34 + off] = (byte) i7;
            buf[36 + off] = '-';
            buf[38 + off] = (byte) (i8 >> 8);
            buf[40 + off] = (byte) i8;
            buf[42 + off] = (byte) (i9 >> 8);
            buf[44 + off] = (byte) i9;
            buf[46 + off] = '-';
            buf[48 + off] = (byte) (i10 >> 8);
            buf[50 + off] = (byte) i10;
            buf[52 + off] = (byte) (i11 >> 8);
            buf[54 + off] = (byte) i11;
            buf[56 + off] = (byte) (i12 >> 8);
            buf[58 + off] = (byte) i12;
            buf[60 + off] = (byte) (i13 >> 8);
            buf[62 + off] = (byte) i13;
            buf[64 + off] = (byte) (i14 >> 8);
            buf[66 + off] = (byte) i14;
            buf[68 + off] = (byte) (i15 >> 8);
            buf[70 + off] = (byte) i15;

            return new String(buf, UTF16);
        }
```

Reply:
We should avoid extra indentation when possible.